remo/sdk.py

```diff
@@ -762,3 +764,44 @@ def view_annotation_stats(self, annotation_set_id: int):
 
     def _view(self, url, *args, **kwargs):
         return self.viewer.browse(self.api.url(url, *args, **kwargs))
+    
+    def generate_annotations_from_folders(self, path: str, output_filename: str):
+        """
+        Returns Annotations file from a given folder of folders where the 
+        name of the folder is the class label and saves it to a CSV in the same 
+        directory.
+
+        Input: path to folder: str
+               desired filename: str
+
+        Returns: 
+                CSV of Files: CSV
+                CSV of data encoding: CSV
+        """
+
+        def natural_sort(l): 
+            convert = lambda text: int(text) if text.isdigit() else text.lower() 
+            alphanum_key = lambda key: [ convert(c) for c in re.split('([0-9]+)', key) ] 
+            return sorted(l, key = alphanum_key)
+        def classes(path):
+            classes = [d.name for d in os.scandir(path) if d.is_dir()]
+            classes = natural_sort(classes)
+            class_to_idx = {cls_name: i for i, cls_name in enumerate(classes)}
+            return classes, class_to_idx
+
+        classes_list, mapping = classes(path)
+
+        im_dict = {}
+        im_dict["file_name"] = "class_name"
+        for class_name in classes_list:
+            im_list = os.listdir(path + "/" + class_name)
+            for im in im_list:
+                im_dict[os.path.abspath(path) + "/" + class_name + "/" + im] = class_name
+
+        with open(output_filename + "_annotations.csv", "w") as f:
+            for key in im_dict.keys():
+                f.write("%s, %s\n" % (key, im_dict[key]))
+
+        with open(output_filename + "_encoding.csv", "w") as g:
```

Comment:
@SreeHarshaNelaturu I'm a bit unclear on this part, I thought we wanted to pass class encoding as an input to the function. Do you mind clarifying?

Reply:
I could introduce a bool here which asks whether to save the mapping or not, my idea is save a mapping if required, otherwise the user can pass their own.

Reply:
I guess the idea of mapping was, that we can map existing folder names to other labels

Reply:
Yes, map folder names -> index. The mapping would look like "dog" : 0, "cat" : 1. This is generated by the function itself, there is an option to avoid generating it that I have added.

Reply:
@drewlr @SreeHarshaNelaturu I'd like to clarify my point:
for example in dataset you have folders: n123, n223, n333, and 1st folder contains 'cats', 2nd - 'dogs', and 3rd - 'flowers'
I'd like to get generated annotation file as:
001.jpg, dog
002.jpg, cat
003.jpg, dog

by providing class encoding dictionary: 
`{
'n123': 'cat',
'n223': 'dog',
'n333': 'flower'
}`

and in case I don't want to pass class encoding information, I can get generated annotation file as:
001.jpg, n223
002.jpg, n123
003.jpg, n223

Reply:
@vmanilo  yes, my understanding of using class encoding was the same as yours. Class encoding to be an optional **input** to the function, which if passed would result in a decoded annotation file. 
 This was what I was asking with the comment :) 

@SreeHarshaNelaturu what do you think of this? I'm not clear on what's happening at the moment...how you can generate the encoding if the user doesn't provide an encoding mapping?
The use case I have in mind is one where folders are originally encoded like "n123", and we want to decode them to "dog".

Reply:
The point in this case is as follows:
I have a set of folders name "dog", "cat", "horse". There is no index/integer labeling that I currently have, and I am not concerned with their exact integer label (the user has no specific encoding) in this case,  a mapping dictionary is generated

{0 : "dog", 1 : "cat", 2 : "horse"} which is the encoding. This is important in case there is no such encoding present on the users end at all.

In case the user does not need this, they can set generate_encoding=False